common/tls/src/main/java/io/helidon/common/tls/ConfiguredTlsManager.java

```diff
@@ -0,0 +1,390 @@
+                                                               + ", but algorithm is not specified");
+                }
+                return SecureRandom.getInstance(tlsConfig.secureRandomAlgorithm().get(), tlsConfig.secureRandomProvider().get());
+            }
+        } catch (GeneralSecurityException e) {
+            throw new IllegalArgumentException("invalid configuration for secure random, cannot create it", e);
+        }
+
+        return RANDOM.get();
+    }
+
+    protected KeyManagerFactory buildKmf(TlsConfig target,
+                                         SecureRandom secureRandom,
+                                         PrivateKey privateKey,
+                                         Certificate[] certificates) {
+        byte[] passwordBytes = new byte[64];
+        secureRandom.nextBytes(passwordBytes);
+        char[] password = Base64.getEncoder().encodeToString(passwordBytes).toCharArray();
+
+        try {
+            KeyStore ks = internalKeystore(target);
+            ks.setKeyEntry("key",
+                           privateKey,
+                           password,
+                           certificates);
+
+            KeyManagerFactory kmf = kmf(target);
+            kmf.init(ks, password);
+            return kmf;
+        } catch (UnrecoverableKeyException | KeyStoreException | NoSuchAlgorithmException e) {
+            throw new IllegalArgumentException("Invalid configuration for key management factory, cannot create factory", e);
+        }
+    }
+
+    /**
+     * Creates an internal keystore and loads it with no password and no data.
+     *
+     * @param tlsConfig TLS config
+     * @return a new keystore
+     */
+    protected KeyStore internalKeystore(TlsConfig tlsConfig) {
+        try {
+            String type = tlsConfig.internalKeystoreType().orElseGet(KeyStore::getDefaultType);
+
+            KeyStore ks;
+            if (tlsConfig.internalKeystoreProvider().isEmpty()) {
+                ks = KeyStore.getInstance(type);
+            } else {
+                ks = KeyStore.getInstance(type, tlsConfig.internalKeystoreProvider().get());
+            }
+
+            ks.load(null, null);
+            return ks;
+        } catch (KeyStoreException
+                 | NoSuchProviderException
+                 | IOException
+                 | NoSuchAlgorithmException
+                 | CertificateException e) {
+            throw new IllegalArgumentException("Invalid configuration of internal keystores. Provider: "
+                                                       + tlsConfig.internalKeystoreProvider()
+                                                       + ", type: " + tlsConfig.internalKeystoreType(), e);
+        }
+    }
+
+    /**
+     * Create a new trust manager factory based on the configuration.
+     *
+     * @param tlsConfig TLS config
+     * @return a new trust manager factory
+     */
+    protected TrustManagerFactory tmf(TlsConfig tlsConfig) {
```

Comment:
@tomas-langer would like to discuss consolidation of `tmf(TlsConfig)`, `buildTmf(TlsConfig)`m and `toTmf(TlsConfig)`; or we I can file a follow-up issue to refactor later...